css/src/components/form/form.scss

```
@@ -0,0 +1,238 @@
+	font-size: $form-font-size;
+	text-align: left;
+
+	&.control-icons {
+		.icon {
+			position: absolute;
+			inset-block-start: 0;
+			inset-inline-start: 0;
+			width: $icon-size;
+			height: $icon-size;
+			color: $icon-color;
+			font-size: $form-font-size;
+			pointer-events: none;
+			z-index: $zindex-multi;
+			padding-inline-start: $icon-size;
+		}
+
+		.input,
+		.select select {
+			padding-inline-start: $icon-size;
+		}
+
+		.input,
+		.select {
+			~ .icon {
+				font-size: $form-font-size;
+			}
+
+			&:focus-visible {
+				~ .icon,
+				.icon {
+					color: $icon-active-color;
+				}
+			}
+
+			&.input-sm,
+			&.select-sm {
+				~ .icon,
+				.icon {
+					font-size: $form-font-size-sm;
+				}
+			}
+
+			&.input-lg,
+			&.select-lg {
+				~ .icon,
+				.icon {
+					font-size: $form-font-size-lg;
+				}
+			}
+		}
+
+		&.control-icon-right {
+			.input {
+				padding-inline: $control-padding-horizontal;
+
+				~ .icon {
+					inset-inline-start: unset;
+					inset-inline-end: 0;
+				}
+			}
+
+			.select {
+				~ .icon {
+					inset-inline-start: unset;
+					inset-inline-end: 2.25em;
+				}
+			}
+
+			.input,
+			.select select {
+				padding-inline-end: $icon-size;
+			}
+
+			.input[type='date'] {
+				position: relative;
+
+				&::-webkit-calendar-picker-indicator {
+					position: absolute;
+					width: 1.5em;
+					background: none;
+					color: transparent;
+					inset-inline-end: 0;
+					z-index: $zindex-dropdown;
+				}
+			}
+		}
+	}
+
+	&.is-loading {
+		&::after {
+			@include loader;
+
+			position: absolute !important;
+			inset-inline-end: 0.625em;
+			inset-block-start: 0.625em;
+			z-index: $zindex-multi;
+		}
+
+		&.control-small::after {
+			font-size: $form-font-size-sm;
+		}
+
+		&.control-medium::after {
+			font-size: $form-font-size;
+		}
+
+		&.control-large::after {
+			font-size: $form-font-size-lg;
+		}
+	}
+}
+
+.field {
+	&:not(:last-child) {
+		margin-block-end: 0.75rem;
+	}
+
+	// Modifiers
+
+	&.field-addons {
+		display: flex;
+		justify-content: flex-start;
+
+		.control {
+			&:not(:last-child) {
+				margin-inline-end: -1px;
+			}
+
+			&:first-child {
+				.button,
+				.input,
+				.select select {
+					border-top-right-radius: 0;
+					border-bottom-right-radius: 0;
+				}
+			}
+
+			&:last-child {
+				.button,
+				.input,
+				.select select {
+					border-top-left-radius: 0;
+					border-bottom-left-radius: 0;
+				}
+			}
+
+			&:not(:first-child):not(:last-child) {
+				.button,
+				.input,
+				.select select {
+					border-radius: 0;
+				}
+			}
+		}
+	}
+
+	.field-label {
+		margin-block-end: $spacer-3;
+
+		@include tablet {
+			flex: 1 0 0;
+
+			&.field-label-sm {
+				font-size: $form-font-size-sm;
+			}
+
+			&.field-label-lg {
+				font-size: $form-font-size-lg;
+			}
+		}
+	}
+
+	&.field-horizontal {
+		@include tablet {
+			display: flex;
+		}
+
+		.field-label {
+			@include tablet {
+				margin-inline-end: $spacer-7;
+				align-self: center;
+				text-align: right;
```

Is this an opportunity to use the logical value `end`?

```suggestion
				text-align: end;
```

---

Updated